src/artkit/model/llm/vertexai/_vertexai.py

```diff
@@ -0,0 +1,233 @@
+        """Placeholder class for missing ``GenerativeModel`` class."""
+
+    class Part(metaclass=MissingClassMeta, module="vertexai.generative_models"):  # type: ignore
+        """Placeholder class for missing ``Part`` class."""
+
+
+#
+# Type variables
+#
+
+T_VertexAIChat = TypeVar("T_VertexAIChat", bound="VertexAIChat")
+
+#
+# Class declarations
+#
+
+
+@inheritdoc(match="""[see superclass]""")
+class VertexAIChat(ChatModelConnector[GenerativeModel], metaclass=ABCMeta):
+    """
+    Base class for Vertex AI LLMs.
+    """
+
+    region: str | None
+    gcp_project_id: str | None
+    #: If ``False``, disable all accessible safety categories; otherwise, enable them.
+    safety: bool
+
+    @classmethod
+    def get_default_api_key_env(cls) -> str:
+        """[see superclass]"""
+        return ""
+
+    def _make_client(self) -> GenerativeModel:  # pragma: no cover
+        """
+        This method handles the authentication and connection to the Vertex AI. It
+        assumes you have followed the instructions in
+        sphinx/source/user_guide/introduction_to_artkit/connecting_to_genai_models.ipynb
+        to setup Application Default Credentials(ADC) to access GCP.
+        """
+        vertexai.init(project=self.gcp_project_id, location=self.region)
+        return GenerativeModel(self.model_id, system_instruction=self.system_prompt)
+
+    def get_model_params(self) -> Mapping[str, Any]:
+        """[see superclass]"""
+        return dict(**super().get_model_params(), safety=self.safety)
+
+    @subsdoc(
+        # The pattern matches the row defining model_params, and move it to the end
+        # of the docstring.
+        pattern=r"(:param model_params: .*\n)((:?.|\n)*\S)(\n|\s)*",
+        replacement=r"\2\1",
+    )
+    @appenddoc(to=ChatModelConnector.__init__)
+    def __init__(
+        self,
+        *,
+        model_id: str,
+        api_key_env: str | None = None,
+        initial_delay: float = 1,
+        exponential_base: float = 2,
+        jitter: bool = True,
+        max_retries: int = 10,
+        system_prompt: str | None = None,
+        region: str | None = None,
+        gcp_project_id: str,
+        safety: bool = True,
+        **model_params: Any,
+    ) -> None:
+        """
+        :param region: The specific GCP region to connect to.
+        :param gcp_project_id: The GCP project ID.
+        :param safety: Safety settings for the model.
+        """
+        super().__init__(
+            model_id=model_id,
+            api_key_env=api_key_env,
+            initial_delay=initial_delay,
+            exponential_base=exponential_base,
+            jitter=jitter,
+            max_retries=max_retries,
+            system_prompt=system_prompt,
+            **model_params,
+        )
+        self.region = region if region else "us-east1"
+        self.gcp_project_id = gcp_project_id
+        self.safety = safety
+
+    def _get_safety_settings(self) -> dict[str, str] | None:
```

Comment:
Is this specifically for Gemini/Google models?

Reply:
yup!